--- a/src/Output/Lines.php
+++ b/src/Output/Lines.php
@@ -79,11 +79,11 @@ class Lines
         $max = $this->lines->map(fn ($col) => collect($col)->count())->max();

         $lines = $this->lines->map(fn ($col) => collect($col))->map(function (Collection $col) use ($max) {
-            $maxLength = $col->max(fn ($l) => mb_strlen(Util::stripEscapeSequences($l)));
+            $maxLength = $col->max(fn ($l) => mb_strwidth(Util::stripEscapeSequences($l)));
             $spaces = str_repeat($this->paddingCharacter, $maxLength ?? 0);

             if (in_array($this->align, ['top', 'bottom'])) {
-                // Make sure all lines have the same length
+                // Make sure all lines have the same visible width
                 $col = $col->map(function ($l) use ($maxLength) {
-                    $length = mb_strlen(Util::stripEscapeSequences($l));
+                    $length = mb_strwidth(Util::stripEscapeSequences($l));

                     if ($length < $maxLength) {
                         $l .= str_repeat($this->paddingCharacter, $maxLength - $length);
